liveman_api/lib/liveman_api.ex

```diff
@@ -1,18 +1,44 @@
 defmodule LivemanApi do
   @moduledoc """
-  Documentation for `LivemanApi`.
+  API Wrapper for Liveman
   """
 
-  @doc """
-  Hello world.
+  @timeout 30_000
 
-  ## Examples
+  def get(base_url, url, query \\ %{}) do
+    base_url
+    |> client()
+    |> Tesla.get(url, query: Map.to_list(query))
+    |> handle_response()
+  end
 
-      iex> LivemanApi.hello()
-      :world
+  def post(base_url, url, body \\ %{}) do
+    base_url
+    |> client()
+    |> Tesla.post(url, body)
+    |> handle_response()
+  end
 
-  """
-  def hello do
-    :world
+  defp client(base_url, opts \\ %{}) do
+    headers = opts[:headers] || []
+
+    middleware = [
+      {Tesla.Middleware.BaseUrl, base_url},
+      Tesla.Middleware.JSON,
+      {Tesla.Middleware.Headers, [{"Content-Type", "application/json"} | headers]},
+      {Tesla.Middleware.Timeout, [timeout: @timeout]}
+    ]
+
+    adapter = {Tesla.Adapter.Hackney, [recv_timeout: @timeout]}
+
+    Tesla.client(middleware, adapter)
   end
+
+  defp handle_response({:ok, %{status: status, body: body}}) when status in 200..299,
+    do: {:ok, body}
```

Comment:
I think later, maybe for improvement, we need to expose the headers as well. It might be useful for API document. 👍